bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+package edu.gemini.ags.impl
+
+import edu.gemini.ags.api.AgsMagnitude.MagnitudeTable
+import edu.gemini.ags.api.AgsStrategy.Estimate
+import edu.gemini.ags.api.{AgsAnalysis, AgsGuideQuality, AgsMagnitude, AgsStrategy}
+import edu.gemini.catalog.api.CatalogQuery
+import edu.gemini.skycalc.Angle
+import edu.gemini.spModel.ags.AgsStrategyKey
+import edu.gemini.spModel.core.SiderealTarget
+import edu.gemini.spModel.guide.{GuideProbe, ValidatableGuideProbe}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.obscomp.SPInstObsComp
+import edu.gemini.spModel.telescope.{PosAngleConstraint, PosAngleConstraintAware}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scalaz._
+import Scalaz._
+
+/**
+ * Represents an AGS strategy consisting of multiple substrategies acting together.
+ */
+sealed case class MultiProbeStrategy(key: AgsStrategyKey, strategies: List[AgsStrategy]) extends AgsStrategy {
+
+  override def magnitudes(ctx: ObsContext, mt: MagnitudeTable): List[(GuideProbe, AgsMagnitude.MagnitudeCalc)] =
+    strategies.flatMap(_.magnitudes(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable): List[AgsAnalysis] =
+    strategies.flatMap(_.analyze(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable, guideProbe: ValidatableGuideProbe, guideStar: SiderealTarget): Option[AgsAnalysis] =
+    AgsAnalysis.analysis(ctx, mt, guideProbe, guideStar, probeBands)
+
+  override def catalogQueries(ctx: ObsContext, mt: MagnitudeTable): List[CatalogQuery] =
+    strategies.flatMap(_.catalogQueries(ctx, mt))
+
+  override def candidates(ctx: ObsContext, mt: MagnitudeTable): Future[List[(GuideProbe, List[SiderealTarget])]] =
+    Future.traverse(strategies)(_.candidates(ctx, mt)).map(_.flatten)
+
+  override def estimate(ctx: ObsContext, mt: MagnitudeTable): Future[AgsStrategy.Estimate] =
+    Future.fold(strategies.map(_.estimate(ctx, mt)))(1.0)((p,est) => p * est.probability).map(Estimate(_))
```

Comment:
Position angle constraint is a factor in estimation so this implementation is not kosher I think.  For example, in `SingleProbeStrategy.estimate`:

````scala
    val pac   = ctx.getPosAngleConstraint(UNBOUNDED)
    val cv    = CandidateValidator(params, mt, candidates)
    val steps = pac.steps(ctx.getPositionAngle, params.stepSize.toOldModel).toList.asScala
    val anglesWithResults  = steps.filter { angle => cv.exists(ctx.withPositionAngle(angle)) }
    val successProbability = anglesWithResults.size.toDouble / steps.size.toDouble
    AgsStrategy.Estimate.toEstimate(successProbability)
````

So it seems like you _do_ have the issue you identified already where one strategy can only find a guide star at one particular angle and another only at a different position angle.  The solution you came up with for the select case seems reasonable.